fw/http_msg.c

```diff
@@ -1496,31 +1496,29 @@ static inline void
 __tfw_h2_msg_move_frags(struct sk_buff *skb, int frag_idx,
 			TfwHttpRespCleanup *cleanup)
 {
-	int i;
-	int len;
+	int i, len;
 	struct page *page;
 	struct skb_shared_info *si = skb_shinfo(skb);
 
 	for (i = 0; i < frag_idx; i++) {
 		page = skb_frag_page(&si->frags[i]);
 		cleanup->pages[i] = compound_head(page);
 		cleanup->pages_sz++;
-		len = skb_frag_size(&si->frags[i]);
-		ss_skb_adjust_data_len(skb, -len);
-		si->nr_frags--;
+		len += skb_frag_size(&si->frags[i]);
```

Comment:
With this change, the `len` variable should be initialized to zero either when it's declared or in the `for` loop initialization, the same way as `i` is initialized. The latter would seem more logical for `len` initialization.